modules/@ergonode/core/src/store/dictionaries/actions.js

```diff
@@ -22,52 +25,90 @@ const modulesDictionaries = Object.values(extendsModules)
+        keys.forEach((key) => {
+            requests.push(async () => {
+                try {
+                    const modulesDictionary = modulesDictionaries.find(
+                        dictionary => dictionary.stateProp === key,
+                    );
 
-            commit('__SET_STATE', {
-                key: stateProp,
-                value,
+                    const stateValue = JSON.stringify(state[key]);
+                    const dictionaryDefaultValue = JSON.stringify(modulesDictionary.defaultValue);
+
+                    if (stateValue === dictionaryDefaultValue) {
+                        const {
+                            request: {
+                                path,
+                                config,
+                            },
+                            dataMapper = response => response,
+                        } = modulesDictionary;
+
+                        const response = await get({
+                            $axios: this.app.$axios,
+                            path,
+                            config,
+                        });
+
+                        commit('__SET_STATE', {
+                            key,
+                            value: dataMapper(response),
+                        });
+                    }
+                } catch (e) {
+                    this.app.$addAlert({
+                        type: ALERT_TYPE.ERROR,
+                        message: 'Dictionary couldn\'t be fetched',
+                    });
+                }
             });
         });
 
-        await Promise.all(promises);
+        await Promise.all(requests.map(request => request()));
     },
     async getDictionary({
         commit,
     }, {
-        dictionaryName,
+        key,
     }) {
-        const {
-            stateProp,
-            requestPath,
-            isGrid = false,
-        } = modulesDictionaries.find(({
-            stateProp: name,
-        }) => name === dictionaryName);
-        const path = `${requestPath}${isGrid ? '?view=list' : ''}`;
-        const response = await get({
-            $axios: this.app.$axios,
-            path,
-        });
-        const value = isGrid ? response.collection : response;
+        try {
+            const modulesDictionary = modulesDictionaries.find(
+                dictionary => dictionary.stateProp === key,
+            );
 
-        commit('__SET_STATE', {
-            key: stateProp,
-            value,
-        });
+            const {
+                request: {
+                    path,
+                    config,
+                },
+                dataMapper = response => response,
+            } = modulesDictionary;
+
+            const response = await get({
+                $axios: this.app.$axios,
+                path,
+                config,
+            });
+
+            commit('__SET_STATE', {
+                key,
+                value: dataMapper(response),
+            });
```

Comment:
use functional programming and move this code into a smaller method and use it in `getDictionary` and `getInitialDictionaries` methods.

Reply:
I do not think it's necessary. We should not reduced duplicated code by wrapping them into functions in case that there is nothing to wrap. Destructuring object along with two methods calls is well documented like it is. Creating function outside vuex action will break it convention - i would need to pass everything context which would not be more readable at all.